kolibri/plugins/learn/assets/src/views/index.vue

```diff
@@ -39,26 +39,28 @@
 
     <div v-if="!isSearchPage">
       <tabs>
-        <tab-link
-          type="icon-and-title"
-          :title="$tr('recommended')"
-          icon="forum"
-          :link="recommendedLink"
-        />
-        <tab-link
-          type="icon-and-title"
-          :title="$tr('topics')"
-          icon="folder"
-          :link="topicsLink"
-        />
-        <tab-link
-          name="exam-link"
-          v-if="isUserLoggedIn && userHasMemberships"
-          type="icon-and-title"
-          :title="$tr('exams')"
-          icon="assignments"
-          :link="examsLink"
-        />
+        <template scope="tab">
+          <component
+            :is="tab.link"
```

Comment:
> I mainly wanted to avoid having complex data structures (e.g. arrays of objects) as props because they're harder to validate and document.
That makes sense, it just seems counter intuitive to the idea behind a child component. We have to pass in the same props using the same grandchildren over and over, which makes the template feel more verbose than it needs to be.

 > Would they actually reduce code repetition?
Yes, because you'd only specify the granchild component once for every item in the array. However, you'd have to pass in the array via a prop and the grandchild component via a slot, making it kind of complex.

> I'm admittedly not the sharpest tool in the shed
Nah. It took me a lot of experimenting and rereading to understand how they worked. Christian had some trouble with it too. Maybe this is reason enough to forego the approach.

Basically, adding `scope` to a slot while it's being used in a parent passes up the scope of the child to the parent (only within the slot) which might be anti-pattern.